File: Creational/Abstract_factory/abstract.cs
public interface IAbstractFactory {
    IAbstractProductA CreateProductA();
    IAbstractProductB CreateProductB();
}

public interface IAbstractProductA { string UsefulFunctionA(); }
public interface IAbstractProductB { string UsefulFunctionB(); }

class ConcreteFactory1 : IAbstractFactory {
    public IAbstractProductA CreateProductA() => new ProductA1();
    public IAbstractProductB CreateProductB() => new ProductB1();
}

class ProductA1 : IAbstractProductA { public string UsefulFunctionA() => "Product A1 result."; }
class ProductB1 : IAbstractProductB { public string UsefulFunctionB() => "Product B1 result."; }

File: Creational/Abstract_factory/abstract.py
class AbstractFactory(ABC):
    @abstractmethod
    def create_product_a(self): pass
    @abstractmethod
    def create_product_b(self): pass

class ConcreteFactory1(AbstractFactory):
    def create_product_a(self): return ProductA1()
    def create_product_b(self): return ProductB1()

class ProductA1:
    def useful_function_a(self): return "Product A1 result."

class ProductB1:
    def useful_function_b(self): return "Product B1 result."